File: src/mrl/views/SelfView.java
/*
 *  "MysteriesRL", Java Roguelike game.
 *  Copyright (C) 2015 by Serg V. Zhdanovskih (aka Alchemist, aka Norseman).
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package mrl.views;

import java.awt.Color;
import java.awt.event.KeyEvent;
import java.awt.event.MouseEvent;
import jzrlib.jterm.JTerminal;
import mrl.core.GlobalData;
import mrl.creatures.Human;
import mrl.creatures.NPCStats;
import mrl.creatures.body.HumanBody;
import mrl.game.Game;
import mrl.maps.Layer;
import mrl.maps.city.City;

/**
 *
 * @author Serg V. Zhdanovskih
 */
public final class SelfView extends SubView
{
    
    public SelfView(BaseView ownerView, JTerminal terminal)
    {
        super(ownerView, terminal);
    }

    @Override
    public final Game getGameSpace()
    {
        return ((MainView) this.fOwnerView).getGameSpace();
    }

    @Override
    protected void updateView()
    {
        this.fTerminal.clear();
        this.fTerminal.setTextBackground(Color.black);
        this.fTerminal.setTextForeground(Color.white);
        UIPainter.drawBox(fTerminal, 0, 0, 159, 79, false);
        
        Game gameSpace = this.getGameSpace();
        Human player = gameSpace.getPlayerController().getPlayer();
        NPCStats stats = player.getStats();
        HumanBody body = (HumanBody) player.getBody();
        City city = gameSpace.getCity();
        Layer layer = (Layer) gameSpace.getPlainMap();
        
        this.fTerminal.setTextForeground(Color.lightGray);
        UIPainter.drawBox(fTerminal, 2, 2, 50, 50, false, "Player");
        this.fTerminal.write(4,  4, "Player name:  " + player.getName(), Color.white);
        this.fTerminal.write(4,  6, "Sex:          " + player.getSex().name(), Color.white);
        this.fTerminal.write(4,  8, "Age:          " + String.valueOf(player.getAge()), Color.white);
        //this.fTerminal.write(4, 10, "Personality:  " + player.getPersonality().getDesc(), Color.white);

        this.fTerminal.write(4, 14, "Strength:     " + String.valueOf(stats.Str), Color.white);
        this.fTerminal.write(4, 16, "Perception:   " + String.valueOf(stats.Per), Color.white);
        this.fTerminal.write(4, 18, "Endurance:    " + String.valueOf(stats.End), Color.white);
        this.fTerminal.write(4, 20, "Charisma:     " + String.valueOf(stats.Chr), Color.white);
        this.fTerminal.write(4, 22, "Intelligence: " + String.valueOf(stats.Int), Color.white);
        this.fTerminal.write(4, 24, "Agility:      " + String.valueOf(stats.Agi), Color.white);
        this.fTerminal.write(4, 26, "Luck:         " + String.valueOf(stats.Luk), Color.white);

        this.fTerminal.write(4, 30, "Level:        " + String.valueOf(stats.Level), Color.white);
        this.fTerminal.write(4, 32, "XP:           " + String.valueOf(stats.getCurXP()) + " / " + String.valueOf(stats.getNextLevelXP()), Color.white);
        this.fTerminal.write(4, 34, "Carry Weight: " + String.valueOf(0) + " / " + String.valueOf(stats.getCarryWeight()), Color.white);
        this.fTerminal.write(4, 36, "HP:           " + String.valueOf(player.getHP()) + " / " + String.valueOf(player.getHPMax()), Color.white);

        this.fTerminal.write(4, 40, "Stamina:      " + body.getAttribute("stamina"), Color.white);
        this.fTerminal.write(4, 42, "Hunger:       " + body.getAttribute("hunger"), Color.white);

        UIPainter.drawBox(fTerminal, 103, 2, 157, 14, false, "City");
        this.fTerminal.write(105,  4, "Name:        " + city.getName(), Color.white);
        this.fTerminal.write(105,  6, "Streets:     " + String.valueOf(city.getStreets().size()), Color.white);
        this.fTerminal.write(105,  8, "Districts:   " + String.valueOf(city.getDistricts().size()), Color.white);
        this.fTerminal.write(105, 10, "Buildings:   " + String.valueOf(city.getBuildings().size()), Color.white);
        this.fTerminal.write(105, 12, "Inhabitants: " + String.valueOf(layer.getCreatures().getCount()), Color.white);
        
        UIPainter.drawBox(fTerminal, 52, 2, 101, 50, false, "Crimes");
        
        UIPainter.drawBox(fTerminal, 2, 52, 157, 78, false, "Description");
        this.fTerminal.write(4, 54, player.getDesc(), Color.white);
    }
    
    @Override
    public void tick()
    {
    }

    @Override
    public void keyPressed(KeyEvent e)
    {
        switch (e.getKeyCode()) {
            case GlobalData.KEY_ESC:
                this.getMainView().setView(ViewType.vtGame);
                break;

            default:
                break;
        }
    }

    @Override
    public void keyTyped(KeyEvent e)
    {
    }

    @Override
    public void mouseClicked(MouseEvent e)
    {
    }

    @Override
    public void mouseMoved(MouseEvent e)
    {
    }
    
    @Override
    public final void show()
    {
    }
}

File: project/Views/SelfView.cs
/*
 *  "PrimevalRL", roguelike game.
 *  Copyright (C) 2015, 2017 by Serg V. Zhdanovskih.
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

using System;
using PrimevalRL.Creatures;
using PrimevalRL.Game;
using PrimevalRL.Maps;
using ZRLib.Engine;

namespace PrimevalRL.Views
{
    public sealed class SelfView : SubView
    {
        public SelfView(BaseView ownerView, Terminal terminal)
            : base(ownerView, terminal)
        {
        }

        public override MRLGame GameSpace
        {
            get {
                return ((MainView)fOwnerView).GameSpace;
            }
        }

        internal override void UpdateView()
        {
            fTerminal.Clear();
            fTerminal.TextBackground = Colors.Black;
            fTerminal.TextForeground = Colors.White;
            fTerminal.DrawBox(0, 0, 159, 79, false);

            MRLGame gameSpace = GameSpace;
            Human player = gameSpace.PlayerController.Player;
            NPCStats stats = player.Stats;
            HumanBody body = (HumanBody)player.Body;
            City city = gameSpace.City;
            Layer layer = (Layer)gameSpace.BaseRealm.PlainMap;

            fTerminal.TextForeground = Colors.LightGray;
            fTerminal.DrawBox(2, 2, 50, 50, false, "Player");
            fTerminal.Write(4, 4, "Player name:  " + player.Name, Colors.White);
            fTerminal.Write(4, 6, "Sex:          " + player.Sex.ToString(), Colors.White);
            fTerminal.Write(4, 8, "Age:          " + Convert.ToString(player.Age), Colors.White);
            //fTerminal.write(4, 10, "Personality:  " + player.getPersonality().getDesc(), Colors.white);

            fTerminal.Write(4, 14, "Strength:     " + Convert.ToString(stats.Str), Colors.White);
            fTerminal.Write(4, 16, "Perception:   " + Convert.ToString(stats.Per), Colors.White);
            fTerminal.Write(4, 18, "Endurance:    " + Convert.ToString(stats.End), Colors.White);
            fTerminal.Write(4, 20, "Charisma:     " + Convert.ToString(stats.Chr), Colors.White);
            fTerminal.Write(4, 22, "Intelligence: " + Convert.ToString(stats.Int), Colors.White);
            fTerminal.Write(4, 24, "Agility:      " + Convert.ToString(stats.Agi), Colors.White);
            fTerminal.Write(4, 26, "Luck:         " + Convert.ToString(stats.Luk), Colors.White);

            fTerminal.Write(4, 30, "Level:        " + Convert.ToString(stats.Level), Colors.White);
            fTerminal.Write(4, 32, "XP:           " + Convert.ToString(stats.CurXP) + " / " + Convert.ToString(stats.NextLevelXP), Colors.White);
            fTerminal.Write(4, 34, "Carry Weight: " + Convert.ToString(0) + " / " + Convert.ToString(stats.CarryWeight), Colors.White);
            fTerminal.Write(4, 36, "HP:           " + Convert.ToString(player.HP) + " / " + Convert.ToString(player.HPMax), Colors.White);

            fTerminal.Write(4, 40, "Stamina:      " + body.GetAttribute("stamina"), Colors.White);
            fTerminal.Write(4, 42, "Hunger:       " + body.GetAttribute("hunger"), Colors.White);

            fTerminal.DrawBox(103, 2, 157, 14, false, "City");
            fTerminal.Write(105, 4, "Name:        " + city.Name, Colors.White);
            fTerminal.Write(105, 6, "Streets:     " + Convert.ToString(city.Streets.Count), Colors.White);
            fTerminal.Write(105, 8, "Districts:   " + Convert.ToString(city.Districts.Count), Colors.White);
            fTerminal.Write(105, 10, "Buildings:   " + Convert.ToString(city.Buildings.Count), Colors.White);
            fTerminal.Write(105, 12, "Inhabitants: " + Convert.ToString(layer.Creatures.Count), Colors.White);

            fTerminal.DrawBox(52, 2, 101, 50, false, "Crimes");

            fTerminal.DrawBox(2, 52, 157, 78, false, "Description");
            fTerminal.Write(4, 54, player.Desc, Colors.White);
        }

        public override void Tick()
        {
        }

        public override void KeyPressed(KeyEventArgs e)
        {
            switch (e.Key) {
                case Keys.GK_ESCAPE:
                    MainView.View = ViewType.vtGame;
                    break;
            }
        }

        public override void KeyTyped(KeyPressEventArgs e)
        {
        }

        public override void MouseClicked(MouseEventArgs e)
        {
        }

        public override void MouseMoved(MouseMoveEventArgs e)
        {
        }

        public override void Show()
        {
        }
    }
}
